pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -289,7 +289,7 @@ func (p *LaunchTemplateProvider) getUserData(ctx context.Context, constraints *v
 func (p *LaunchTemplateProvider) getBottlerocketUserData(ctx context.Context, constraints *v1alpha1.Constraints, additionalLabels map[string]string, caBundle *string) string {
 	userData := fmt.Sprintf("[settings.kubernetes]\ncluster-name = \"%s\"\napi-server = \"%s\"\n", injection.GetOptions(ctx).ClusterName, injection.GetOptions(ctx).ClusterEndpoint)
 	if constraints.KubeletConfiguration.ClusterDNS != nil {
```

Comment:
to make this more apparent, can this be:

```suggestion
	if len(constraints.KubeletConfiguration.ClusterDNS) > 0 { 
```

Reply:
Great point. I updated the PR with your suggestion.